homeassistant/components/light/homekit_controller.py

```diff
@@ -0,0 +1,182 @@
+                                                   self._brightness / 100)
+            return (red * 255, green * 255, blue * 255)
+        return (255, 255, 255)
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        if self._features & SUPPORT_COLOR_TEMP:
+            return self._color_temperature
+        return 3000
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return self._features
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def assumed_state(self):
+        """Return the assumed state."""
+        return True
+
+    def turn_on(self, **kwargs):
+        """Turn the specified light on."""
+        self._state = True
+
+        rgb = kwargs.get(ATTR_RGB_COLOR)
+        temperature = kwargs.get(ATTR_COLOR_TEMP)
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        characteristics = []
+        if rgb is not None:
+            hue, saturation, value = colorsys.rgb_to_hsv(rgb[0] / 255,
+                                                         rgb[1] / 255,
+                                                         rgb[2] / 255)
+            characteristics.append({'aid': self._aid,
+                                    'iid': self._chars['hue'],
+                                    'value': hue * 360})
+            characteristics.append({'aid': self._aid,
+                                    'iid': self._chars['saturation'],
+                                    'value': saturation * 100})
+            characteristics.append({'aid': self._aid,
+                                    'iid': self._chars['brightness'],
+                                    'value': int(value * 100)})
+        if brightness is not None:
+            characteristics.append({'aid': self._aid,
+                                    'iid': self._chars['brightness'],
+                                    'value': int(brightness * 100 / 255)})
+
+        if temperature is not None:
+            characteristics.append({'aid': self._aid,
+                                    'iid': self._chars['color-temperature'],
+                                    'value': int(temperature)})
+        characteristics.append({'aid': self._aid,
+                                'iid': self._chars['on'],
+                                'value': True})
+        body = json.dumps({'characteristics': characteristics})
+        self._securecon.put('/characteristics', body)
+        self.update_state()
+
+    def turn_off(self, **kwargs):
+        """Turn the specified light off."""
+        characteristics = [{'aid': self._aid,
+                            'iid': self._chars['on'],
+                            'value': False}]
+        body = json.dumps({'characteristics': characteristics})
+        self._securecon.put('/characteristics', body)
+        self.update_state()
```

Comment:
You have should_poll set to True. That means that Home Assistant will call `update()` after it has called a service. So you can remove this statement if you rename the `update_state` to `update`